code/scripting/scripting.h

```diff
@@ -134,6 +134,7 @@ struct script_condition
 {
 	ConditionalType condition_type = CHC_NONE;
 	SCP_string condition_string;
+	int condition_reference_index;
```

Comment:
Should be explicitly initialized to a sane value. `0`? `-1`? Not really sure what "sane" means here.

Reply:
The idea was to use -1 as a 'not set' signifier since this is typically going to be an item in an array or enum. That was decided when I hadn't realized it'd be viable to just evaluate on add and was planning to have them evaluated the first time the condition was checked, though.

Reply:
It still needs to be initialized to _something_ otherwise we have unintialized data here and that is never a good thing.

Reply:
Got it, I misread you initially. Despite the example right above that it slipped my mind that initializing in the struct declaration like that was an option, so I did it locally at what I believe is the only place script_conditions are created, but it'd be better to move it and I will.